PlayerSkeleton.java

```diff
@@ -1,22 +1,217 @@
 
+import java.util.Arrays;
+
 public class PlayerSkeleton {
 
+	private int[][] pWidth;
+	private int[][] pHeight;
+	private int[][][] pBottom;
+	private int[][][] pTop;
+
 	//implement this function to have a working system
 	public int pickMove(State s, int[][] legalMoves) {
-		// System.out.println("Heuristic score = " + Heuristic.evaluate(s));
-		return 0;
+		
+		//Step 1. Get a copy of the current board configuration.
+		int[][] currentField = copyArray(s.getField());
+
+		//Step 2. Test all possible moves of the piece on the board and choose the move with the best heuristic score 
+		int best = tryPossibleMoves(currentField, legalMoves, s);
+
+		return best;
+	}
+
+	/**
+	* This method test all possible moves of the piece and returns the move with the best heuristic score
+	* @param currentField (current board config), legalMoves (possible moves for current piece), s (current board configuration)
+	* @return move with best (highest) heuristic score
+	**/
+	private int tryPossibleMoves(int[][] currentField, int[][] legalMoves, State s) {
```

Comment:
Then this would just be a for loop evaluating each possible next State at each iteration, and then outputing the move with the best evaluation score.